src/test/container-templates/containerTemplatesOCI.test.ts

```diff
@@ -0,0 +1,55 @@
+import { createPlainLog, LogLevel, makeLog } from '../../spec-utils/log';
+import * as assert from 'assert';
+export const output = makeLog(createPlainLog(text => process.stdout.write(text), () => LogLevel.Trace));
+import { fetchTemplate, SelectedTemplate } from '../../spec-configuration/containerTemplatesOCI';
+import * as path from 'path';
+import { readLocalFile } from '../../spec-utils/pfs';
+
+describe('fetchTemplate', async function () {
+	this.timeout('120s');
+
+	it('succeeds on docker-from-docker template', async () => {
+
+		// https://github.com/devcontainers/templates/tree/main/src/docker-from-docker
+		const selectedTemplate: SelectedTemplate = {
+			id: 'ghcr.io/devcontainers/templates/docker-from-docker:latest',
+			options: {'installZsh': 'false', 'upgradePackages': 'true', 'dockerVersion': '20.10', 'moby': 'true', 'enableNonRootDocker': 'true' }
+		};
+
+		const dest = path.relative(process.cwd(), path.join(__dirname, 'tmp1'));
+		const files = await fetchTemplate(output, selectedTemplate, dest);
+		assert.ok(files);
+		// Should only container 1 file '.devcontainer.json'.  The other 3 in this repo should be ignored.
+		assert.strictEqual(files.length, 1);
+
+		// Read File
+		const file = (await readLocalFile(path.join(dest, files[0]))).toString();
+		assert.match(file, /"name": "Docker from Docker"/);
+		assert.match(file, /"installZsh": "false"/);
+		assert.match(file, /"upgradePackages": "true"/);
+		assert.match(file, /"version": "20.10"/);
+		assert.match(file, /"moby": "true"/);
+		assert.match(file, /"enableNonRootDocker": "true"/);
+	});
+
+	it('succeeds on anaconda-postgres template', async () => {
+
+		// https://github.com/devcontainers/templates/tree/main/src/anaconda-postgres
+		const selectedTemplate: SelectedTemplate = {
+			id: 'ghcr.io/devcontainers/templates/anaconda-postgres:latest',
+			options: { 'nodeVersion': 'lts/*' }
+		};
+
+		const dest = path.relative(process.cwd(), path.join(__dirname, 'tmp2'));
+		const files = await fetchTemplate(output, selectedTemplate, dest);
+		assert.ok(files);
+		// Expected:
+		// ./environment.yml, ./.devcontainer/.env, ./.devcontainer/Dockerfile, ./.devcontainer/devcontainer.json, ./.devcontainer/docker-compose.yml, ./.devcontainer/noop.txt
+		assert.strictEqual(files.length, 6);
+
+		// Read File
+		const file = (await readLocalFile(path.join(dest, '.devcontainer', 'Dockerfile'))).toString();
+		assert.match(file, /ARG NODE_VERSION="lts\/\*"/);
```

Comment:
Should this be ⬇️  instead?

```suggestion
		assert.match(file, /ARG NODE_VERSION="lts/*"/);
```

I was mentioning that we probably need to escape chars when we are passing this option.value to the replacement engine. However, the final files should have normal values (ie `lts/*`) ?

Because I am not sure how the dockerfile might parse this?

![image](https://user-images.githubusercontent.com/24955913/195702929-ba37d610-e43c-4189-80ab-5e797d75effb.png)

For tests action [here](https://github.com/devcontainers/templates/blob/main/.github/workflows/test-pr.yaml#L94), we pass escaped value to `sed` but it ends up replacing without the esc chars - which seems correct to me

Reply:
the second argument to `match` above is itself a regex. 

Here are the CLI trace logs substituting this value:
```
2022-10-13T20:10:12.208Z] Scanning file './.devcontainer/.env'
[2022-10-13T20:10:12.209Z] Scanning file './.devcontainer/Dockerfile'
[2022-10-13T20:10:12.209Z] Replacing nodeVersion with lts/*
[2022-10-13T20:10:12.210Z] Scanning file './.devcontainer/devcontainer.json'
[2022-10-13T20:10:12.211Z] Scanning file './.devcontainer/docker-compose.yml'
[2022-10-13T20:10:12.212Z] Scanning file './.devcontainer/noop.txt'
```

And the output of the this test looks good to me:
<img width="558" alt="image" src="https://user-images.githubusercontent.com/23246594/195703560-c14d5013-f4f6-4e98-ae21-a55e9a2d5549.png">

Reply:
Awesome! 🎉